packages/kit/src/core/sync/create_manifest_data/index.spec.js

```diff
@@ -320,6 +320,53 @@ test('allows rest parameters inside segments', () => {
 	]);
 });
 
+test('optional parameters', () => {
```

Comment:
It could be worth adding a case like `/[[locale]]/[...slug] `. I'm assuming `locale` should be greedy, but it'd be good to verify that with a test. You'd probably want to use a matcher with it in any real situation